Saule/JsonApiSerializerOfT.cs

```diff
@@ -93,8 +94,9 @@ public JToken Serialize(object @object, Uri requestUri, JsonApiConfiguration con
             var queryContext = GetQueryContext(request.GetQueryNameValuePairs());
 
             _serializer.QueryContext = queryContext;
+            var apiResourceProvider = new DefaultApiResourceProvider(new T());
```

Comment:
Since this method gets a `JsonApiConfiguration` passed in, shouldn't it use the factory from there? I see why that's hard (no http request message), but it does make sense... Something to think about anyway.

Reply:
that factory depends on `ReturnsResourceAttribute` that places `ApiResource` to `HttpRequestMessage.Properties`. but since we don't have it for these raw cases of `JsonApiSerializer` i went with creating it as is. I can also update the code to add `ApiResource` to `HttpRequestMessage.Properties` and then factory would work as well

Reply:
I think it would be confusing to ignore the provided factory, so I guess it would be fine to create a dummy http request message. It's confusing either way, but seems less confusing than to ignore the provided factory.

Reply:
ok, makes sense. i'll update it

Reply:
added a new internal method in ReturnsResource and used it in both places